### PR TITLE
[Dark mode] Toggle

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

Allow in-app toggle to switch from light to dark mode

## How

Helped by this [stackoverflow thread](https://stackoverflow.com/questions/63011773/how-to-customize-both-dark-and-light-themes-in-storybook-6-0)

Use [this addon](https://storybook.js.org/addons/storybook-dark-mode) to toggle Storybook mode
Update Storybook theming to use our DS color tokens
Tweak it to update stories theme
Keep URL query toggle needed for e2e testing

## Snapshot

image

### CODE REVIEW

File: packages/strapi-design-system/.storybook/hooks/useThemeType.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import React, { useState, useEffect } from 'react';
+import { parse } from 'qs';
+import addons from '@storybook/addons';
+
+const channel = addons.getChannel()
+
+export const useThemeType = () => {
+  const [isDark, setDark] = useState(false);
```
Comment (HichamELBSI) [inline packages/strapi-design-system/.storybook/hooks/useThemeType.js:8]: Can it be sync with a local storage?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/strapi-design-system/.storybook/hooks/useThemeType.js (hunk 1 above)
Comment (HichamELBSI) [reply packages/strapi-design-system/.storybook/hooks/useThemeType.js:8]: Yes. we can do that in a second iteration 👌

**Review by HichamELBSI — APPROVED** [review]
lgtm

File: packages/strapi-design-system/.storybook/hooks/useThemeType.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/strapi-design-system/.storybook/hooks/useThemeType.js:8]: nice idea indeed! 
will add a card in growth backlog to not forget it 👌

**Pull request merged**